.githooks/pre-commit

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+
+changed_files=$(git diff --cached --name-only --diff-filter=M)
```

Comment:
I am not sure all markdown files should have a table of contents. In particular, the pages under `./docs/book/` don't need them.

Reply:
For each md file is for an independent topic, it is fine to have this table for each md file.
For the md files in book, I think it is the same situation. But some maintainers review will be needed.

Reply:
Besides the md file in the docs, should we also mark the readme in the root path?